process_data.py

```diff
@@ -158,67 +185,108 @@ def check_for_missing_data(hours):
     # reading the data of all grid points one by one takes up a lot of CPU. Therefore, the dataset is analysed in
     # pieces: the subsets are read and processed consecutively.
     n_subsets = int(np.ceil(float(len(lats)) / read_n_lats_at_once))
 
-    for i_subset in range(n_subsets):
+    # Choose subsets to be processed in this run
+    subset_range = get_subset_range_from_input(input_subset_ids, n_subsets)
+    
+    # Loop over all specified subsets to write processed data to the output file.
+    counter = 0
+    total_iters = len(lats) * len(lons)*len(subset_range)/n_subsets
+    start_time = timer()
+
+    for i_subset in subset_range:
+        # Find latitudes corresponding to the current i_subset
         i_lat0 = i_subset * read_n_lats_at_once
         if i_lat0+read_n_lats_at_once < len(lats):
-            lats_subset = range(i_lat0, i_lat0 + read_n_lats_at_once)
+            lat_ids_subset = range(i_lat0, i_lat0 + read_n_lats_at_once)
         else:
-            lats_subset = range(i_lat0, len(lats))
+            lat_ids_subset = range(i_lat0, len(lats))
+        lats_subset = lats[lat_ids_subset]
+        print("Subset {}, Latitude(s) analysed: {} to {}".format(i_subset, lats_subset[0], lats_subset[-1]))
+
+        # Initialize result arrays for this subset
+        res = initialize_result_arrays(lats_subset, lons)
 
-        v_levels_east = ds.variables['u'][:, i_highest_level:, lats_subset, :].values
-        v_levels_north = ds.variables['v'][:, i_highest_level:, lats_subset, :].values
+        print('    Result array configured, reading subset input now, time lapsed: {:.2f} hrs'.format(float(timer()-start_time)/3600))
+
+        # Read data for the subset latitudes  
+        v_levels_east = ds.variables['u'][:, i_highest_level:, lat_ids_subset, :].values
+        v_levels_north = ds.variables['v'][:, i_highest_level:, lat_ids_subset, :].values
         v_levels = (v_levels_east**2 + v_levels_north**2)**.5
 
-        t_levels = ds.variables['t'][:, i_highest_level:, lats_subset, :].values
-        q_levels = ds.variables['q'][:, i_highest_level:, lats_subset, :].values
+        t_levels = ds.variables['t'][:, i_highest_level:, lat_ids_subset, :].values
+        q_levels = ds.variables['q'][:, i_highest_level:, lat_ids_subset, :].values
 
         try:
-            surface_pressure = ds.variables['sp'][:, lats_subset, :].values
+            surface_pressure = ds.variables['sp'][:, lat_ids_subset, :].values
         except KeyError:
-            surface_pressure = np.exp(ds.variables['lnsp'][:, lats_subset, :].values)
+            surface_pressure = np.exp(ds.variables['lnsp'][:, lat_ids_subset, :].values)
+
+        print('    Input read, performing satistical analysis now, time lapsed: {:.2f} hrs'.format(float(timer()-start_time)/3600))
 
-        for row_in_v_levels, i_lat in enumerate(lats_subset):
+        for i_lat_in_subset in range(len(lat_ids_subset)):# All subsets are written out, always start at 0 for a new subset
```

Comment:
with written out, you mean that a separate file is saved per subset? please rephrase

Reply:
Edited in the style corrections: 33eef43